pkg/config/tlscfg/cert_watcher_test.go

```diff
@@ -373,16 +386,23 @@ func TestAddCertsToWatch_err(t *testing.T) {
 			opts: Options{
 				CAPath:       caCert,
 				ClientCAPath: caCert,
-				CertPath:     serverCert,
 				KeyPath:      "doesnotexists",
```

Comment:
I edited this test case because when both CertPath and KeyPath are non-empty, [LoadX509KeyPair is called and returns an error and a nil certWatcher](https://github.com/jaegertracing/jaeger/blob/1d3bc1dc6334ce6b09546d9ddfadc5765820a5e2/pkg/config/tlscfg/cert_watcher.go#L56-L59). That was why I got a [nil pointer dereference when called watchCertPair outside certWatcher constructor](https://github.com/jaegertracing/jaeger/pull/4260#issuecomment-1453968676).

I think there's no point testing LoadX509KeyPair and getting a nil certWatcher here. So delete CertPath in this case to let watchCertPair tested.

Reply:
this just tells me that nil pointer error is introduced in this change, not that the test needs changing. I made comment elsewhere that you're not handling / propagating errors properly.